File: behavior_tree_leaves/example_nodes/cpp/condition_example.cpp
#include <ros/ros.h>
#include <actionlib/server/simple_action_server.h>//actionlib
#include <behavior_tree_core/BTAction.h>




enum Status {RUNNING,SUCCESS, FAILURE};


class BTAction
{
protected:

  ros::NodeHandle nh_;
  // NodeHandle instance must be created before this line. Otherwise strange errors may occur.
  actionlib::SimpleActionServer<behavior_tree_core::BTAction> as_;
  std::string action_name_;
  // create messages that are used to published feedback/result
  behavior_tree_core::BTFeedback feedback_;
  behavior_tree_core::BTResult result_;



public:

  BTAction(std::string name) :
    as_(nh_, name, boost::bind(&BTAction::execute_callback, this, _1), false),
    action_name_(name)
  {
 //start the action server (action in sense of Actionlib not BT action)
    as_.start();
     ROS_INFO("Condition Server Started");


  }

  ~BTAction(void)
  {

  }
  void execute_callback(const behavior_tree_core::BTGoalConstPtr &goal)
  {
    if(true){
    set_status(SUCCESS);
    }else{
    set_status(FAILURE);
    }
  }



  //returns the status to the client (Behavior Tree)
    void set_status(int status){
        //Set The feedback and result of BT.action
        feedback_.status = status;
        result_.status = feedback_.status;
        // publish the feedback
        as_.publishFeedback(feedback_);
        // setSucceeded means that it has finished the action (it has returned SUCCESS or FAILURE).
        as_.setSucceeded(result_);

        switch(status){//Print for convenience
        case SUCCESS:
          ROS_INFO("Condition %s Succeeded", ros::this_node::getName().c_str() );
          break;
        case FAILURE:
            ROS_INFO("Condition %s Failed", ros::this_node::getName().c_str() );
          break;
        default:
          break;
        }
    }
  };

int main(int argc, char** argv)
{
    ros::init(argc, argv, "condition");
      ROS_INFO(" Enum: %d",RUNNING);
      ROS_INFO(" condition Ready for Ticks");
  BTAction bt_action(ros::this_node::getName());
  ros::spin();

  return 0;
}

File: behavior_tree_leaves/example_nodes/python/condition_example.py
#! /usr/bin/env python


import rospy

import actionlib

import behavior_tree_core.msg




class BTAction(object):
  # create messages that are used to publish feedback/result
  _feedback = behavior_tree_core.msg.BTFeedback()
  _result   = behavior_tree_core.msg.BTResult()

  def __init__(self, name):
    self._action_name = name
    self._as = actionlib.SimpleActionServer(self._action_name, behavior_tree_core.msg.BTAction, execute_cb=self.execute_cb, auto_start = False)
    self._as.start()
    
  def execute_cb(self, goal):
    
    # publish info to the console for the user
    rospy.loginfo('Checking Condition')
    
    self.set_status('FAILURE')

    

  def set_status(self,status):
      if status == 'SUCCESS':
        self._feedback.status = 1
        self._result.status = self._feedback.status
        rospy.loginfo('Action %s: Succeeded' % self._action_name)
        self._as.set_succeeded(self._result)
      elif status == 'FAILURE':
        self._feedback.status = 2
        self._result.status = self._feedback.status
        rospy.loginfo('Action %s: Failed' % self._action_name)
        self._as.set_succeeded(self._result)
      else:
        rospy.logerr('Action %s: has a wrong return status' % self._action_name)



if __name__ == '__main__':
  rospy.init_node('condition')
  BTAction(rospy.get_name())
  rospy.spin()
